tck/src/main/java/org/eclipse/microprofile/lra/tck/service/LRAMetricService.java

```diff
@@ -28,49 +30,107 @@
-            participantMap.values().forEach(metric -> result.addAndGet(metric.get(name))));
+        metricsPerLra.values().forEach(participantMap ->
+            participantMap.values().forEach(metric -> result.addAndGet(metric.get(metricType))));
 
         return result.get();
     }
 
-    public int getMetric(LRAMetricType name, URI lraId) {
-        return getMetric(name, lraId, ALL);
+    /**
+     * Returns count number for particular metric type filtered by LRA id and the participant class
+     * which defines the participant name (the fully qualified class name is used for it).
+     *
+     * @param metricType  counter for which the metric type will be returned
+     * @param lraId  counter for which lra id will be returned
+     * @param participantClazz  counter for which the participant will be returned
+     * @return metric counter defined based on the method parameters
+     */
+    public int getMetric(LRAMetricType metricType, URI lraId, Class<?> participantClazz) {
+        return getMetric(metricType, lraId, participantClazz.getName());
     }
 
-    public int getMetric(LRAMetricType metric, URI lraId, String participant) {
-        if (metrics.containsKey(lraId) && metrics.get(lraId).containsKey(participant)) {
-            return metrics.get(lraId).get(participant).get(metric);
+    /**
+     * Returns count number for particular metric type filtered by LRA id and the participant's name.
+     * It's expected that the participant name is defined as fully qualified participant class name.
+     *
+     * @param metricType  counter for which metric type will be returned
+     * @param lraId  counter for which lra id will be returned
+     * @param participantClassName  counter for which the participant name will be returned
+     * @return metric counter defined based on the method parameters
+     */
+    public int getMetric(LRAMetricType metricType, URI lraId, String participantClassName) {
+        if (metricsPerLra.containsKey(lraId) && metricsPerLra.get(lraId).containsKey(participantClassName)) {
+            return metricsPerLra.get(lraId).get(participantClassName).get(metricType);
         } else {
-            return -1;
+            return 0;
         }
     }
 
+    /**
+     * Clear the metric storage as whole.
+     */
     public void clear() {
-        metrics.clear();
+        metricsPerLra.clear();
     }
-    
+
+    /**
+     * TODO if the current PR is acceptable then delete the old isLRAFinished method
+     * (which tests whether an LRA is active by making an attempt to enlist with it
+     * which some spec implementations report as a stack trace WARNING if the LRA is
+     * no longer active).
+     *
+     * @param lraId the LRA id to test
+     * @param participantClazz class of the participant that the metrics parameter applies to
+     * @return whether or not an LRA has finished
+     */
+    boolean isLRAFinished(URI lraId, Class<?> participantClazz) {
```

Comment:
I am not sure about this approach. This will require that each test that wants to assert the finish of LRA then it must provide AfterLRA method which must remember to invoke the correct "end" metric. Yes, this can work, but we must remember to keep an eye on new tests (I verified that your current PR passes the TCK). Also, this prohibits the simple tests without any (even AfterLRA) enlistments which may also be required. I believe that https://github.com/eclipse/microprofile-lra/issues/298 should still be implemeted.

Reply:
ad #298) yes, I think this should still be implemented. Just I haven't touched that as I didn't know if you are working on it. I can take care of it (but I don't remember details troubles here)
ad approach) I think it's fine just force this method to use the `AfterLRAParticipant` as the "`participantClazz`" that will be checked here all the time. My refactoring didn't go anyhow far from what was used until now as the client call (before my refactoring) did
```
assertTrue("testAfterLRAParticipant: LRA did not finish",
lraClient.isLRAFinished(lra, lraMetricService, AfterLRAParticipant.class.getName()));
```
Should I freeze this assertion method to the `AfterLRAParticipant`, correct?

Reply:
I will create PR for #298 just facing some issues in our implementation. And no, I don't think that any change to the current code is needed. You don't need to hardcode the AfterLRAParticipant for now.